scripts/exregional_run_prdgen.sh

```diff
@@ -0,0 +1,437 @@
+#
+print_info_msg "
+========================================================================
+Entering script:  \"${scrfunc_fn}\"
+In directory:     \"${scrfunc_dir}\"
+This is the ex-script for the task that runs the post-processor (UPP) on
+the output files corresponding to a specified forecast hour.
+========================================================================"
+#
+#-----------------------------------------------------------------------
+#
+# Set OpenMP variables.
+#
+#-----------------------------------------------------------------------
+#
+export KMP_AFFINITY=${KMP_AFFINITY_RUN_PRDGEN}
+export OMP_NUM_THREADS=${OMP_NUM_THREADS_RUN_PRDGEN}
+export OMP_STACKSIZE=${OMP_STACKSIZE_RUN_PRDGEN}
+#
+#-----------------------------------------------------------------------
+#
+# Load modules.
+#
+#-----------------------------------------------------------------------
+#
+eval ${PRE_TASK_CMDS}
+
+nprocs=$(( NNODES_RUN_PRDGEN*PPN_RUN_PRDGEN ))
+if [ -z "${RUN_CMD_PRDGEN:-}" ] ; then
+  print_err_msg_exit "\
+  Run command was not set in machine file. \
+  Please set RUN_CMD_PRDGEN for your platform"
+else
+  print_info_msg "$VERBOSE" "
+  All executables will be submitted with command \'${RUN_CMD_PRDGEN}\'."
+fi
+#
+#-----------------------------------------------------------------------
+#
+# Get the cycle date and hour (in formats of yyyymmdd and hh, respectively)
+# from CDATE.
+#
+#-----------------------------------------------------------------------
+#
+yyyymmdd=${PDY}
+hh=${cyc}
+#-----------------------------------------------------------------------
+#
+# A separate ${post_fhr} forecast hour variable is required for the post
+# files, since they may or may not be three digits long, depending on the
+# length of the forecast.
+#
+#-----------------------------------------------------------------------
+#
+
+len_fhr=${#fhr}
+if [ ${len_fhr} -eq 9 ]; then
+  post_min=${fhr:4:2}
+  if [ ${post_min} -lt 15 ]; then
+    post_min=00
+  fi
+else
+  post_min=00
+fi
+
+subh_fhr=${fhr}
+if [ ${len_fhr} -eq 2 ]; then
+  post_fhr=${fhr}00
+elif [ ${len_fhr} -eq 3 ]; then
+  if [ "${fhr:0:1}" = "0" ]; then
+    post_fhr="${fhr:1}00"
+  else
+    post_fhr=${fhr}00
+  fi
+elif [ ${len_fhr} -eq 9 ]; then
+  if [ "${fhr:0:1}" = "0" ]; then
+    if [ ${post_min} -eq 00 ]; then
+      post_fhr="${fhr:1:2}00"
+      subh_fhr="${fhr:0:3}"
+    else
+      post_fhr="${fhr:1:2}${fhr:4:2}"
+    fi
+  else
+    if [ ${post_min} -eq 00 ]; then
+      post_fhr="${fhr:0:3}00"
+      subh_fhr="${fhr:0:3}"
+    else
+      post_fhr="${fhr:0:3}${fhr:4:2}"
+    fi
+  fi
+else
+  print_err_msg_exit "\
+The \${fhr} variable contains too few or too many characters:
+  fhr = \"$fhr\""
+fi
+
+# replace fhr with subh_fhr
+echo "fhr=${fhr} and subh_fhr=${subh_fhr}"
+fhr=${subh_fhr}
+
+#
+# background files
+#
+prslev=${COMOUT}/${NET}.${cycle}${dot_ensmem}.prslev.f${fhr}.${POST_OUTPUT_DOMAIN_NAME}.grib2
```

Comment:
Where does `dot_ensmem` get set? Apologies if I missed that one.

Reply:
@christinaholtNOAA It looks like `dot_ensmem` is set in `ush/job_preamble.sh` in the current `develop` branch:

https://github.com/ufs-community/ufs-srweather-app/blob/b34ab05a13d1fc571afd5f99f790e5ec71bc9a14/ush/job_preamble.sh#L27-L31